lib/crypto/signature.js

```diff
@@ -293,7 +293,11 @@ Signature.prototype.hasDefinedHashtype = function() {
   }
   // accept with or without Signature.SIGHASH_ANYONECANPAY by ignoring the bit
   var temp = this.nhashtype & ~Signature.SIGHASH_ANYONECANPAY;
-  if (temp < (Signature.SIGHASH_ALL|Signature.SIGHASH_FORKID) || temp > (Signature.SIGHASH_SINGLE|Signature.SIGHASH_FORKID)) {
+
+  // accept with or without Signature.SIGHASH_FORKID by ignoring it too
```

Comment:
Why does this bit have to be ignored? Wouldn't it necessarily be present all the time?